mne/io/egi/egimff.py

```diff
@@ -686,3 +684,175 @@ def _read_segment_file(self, data, idx, fi, start, stop, cals, mult):
+    evoked(s) objects.
+
+    See Also
+    --------
+    Evoked, EvokedArray, create_info
+    """
+    # Confirm `fname` is a path to an MFF file
+    assert fname.endswith('.mff'), 'fname must be an MFF file.'
+    # Eventually we will want to add a check here to ensure the file is
+    # averaged. This will require parsing of history.xml.
+    # Check for categories.xml file
+    assert op.exists(op.join(fname, 'categories.xml')),\
+        'categories.xml not found in MFF directory. \
+        %s may not be an averaged MFF file.' % fname
+    if condition is None:
+        cats = Reader(fname).categories.categories
+        condition = list(cats.keys())
+    elif not isinstance(condition, list):
+        condition = [condition]
+    logger.info('Reading %s evoked datasets from %s ...'
+                % (len(condition), fname))
+    output = [_read_evoked_mff(fname, c, channel_naming=channel_naming,
+                               verbose=verbose).apply_baseline(baseline)
+              for c in condition]
+    return output if len(output) > 1 else output[0]
+
+
+def _read_evoked_mff(fname, condition, channel_naming='E%d', verbose=None):
+    """Read evoked data from MFF file."""
+    mff_info = _read_header(fname)
+    mff = Reader(fname)
+    cats = mff.categories.categories
+
+    if type(condition) == str:
+        # Condition is interpreted as category name
+        target_cat = condition
+        target_epoch = mff.epochs[target_cat]
+    elif type(condition) == int:
+        # Condition is interpreted as epoch index
+        target_epoch = mff.epochs[condition]
+        target_cat = target_epoch.name
+    else:
+        raise TypeError('Condition must be either int or str.')
+    assert target_cat in cats.keys(), 'Condition "%s" not found in \
+                                      available conditions %s.'\
+                                      % (target_cat, cats.keys())
+
+    # Read in signals from the target epoch
+    data = mff.get_physical_samples_from_epoch(target_epoch)
+    eeg_data, t0 = data['EEG']
+    if 'PNSData' in data.keys():
+        pns_data, t0 = data['PNSData']
+        all_data = np.vstack((eeg_data, pns_data))
+        ch_types = mff_info['chan_type'] + mff_info['pns_types']
+    else:
+        all_data = eeg_data
+        ch_types = mff_info['chan_type']
+    all_data *= 1e-6 # convert to volts
+
+    # Load metadata into info object
+    ### Exclude info['meas_date'] because record time info in
+    ### averaged MFF is the time of the averaging, not true record time.
+    ### We can populate info['custom_ref_applied'] with info from
+    ### info1.xml, but not sure what to put for type int.
+    ch_names = [channel_naming % (i + 1) for i in
+                range(mff.num_channels['EEG'])]
+    ch_names.extend(mff_info['pns_names'])
+    info = create_info(ch_names, mff.sampling_rates['EEG'], ch_types)
+    info['nchan'] = sum(n for n in mff.num_channels.values())
```

Comment:
Do you need the list comprehension here or can you just do `sum(mff.num_channels.values())`?

Reply:
Nice, yeah it doesn't need the list comprehension.